torchbenchmark/models/deep_reinforcement_learning/__init__.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import torch
+from ...util.model import BenchmarkModel
+from torchbenchmark.tasks import REINFORCEMENT_LEARNING
+
+from .config import DQNConfig
+from .env import Env
+from .agent import Agent
+from .model import DQN
+from .memory import ReplayMemory
+
+class Model(BenchmarkModel):
+    task = REINFORCEMENT_LEARNING.OTHER_RL
+
+    def __init__(self, device=None, jit=False):
+        super().__init__()
+        self.device = device
+        self.jit = jit
+        self.args = DQNConfig(self.device)
+        if device == "cuda":
+            torch.cuda.manual_seed(np.random.randint(1, 10000))
+        self.env = Env(self.args)
+        self.env.train()
+        self.dqn = Agent(self.args, self.env)
+        self.mem = ReplayMemory(self.args, self.args.memory_capacity)
+
+    def get_module(self):
+        model = self.dqn.online_net
+        state = self.env.reset()
+        return model, (state.unsqueeze(0), )
+
+    def train(self, niter = 1):
+        if self.jit:
+            return NotImplemented
+        self.dqn.train()
+        T, done = 0, True
+        for T in range(1, niter + 1):
+            if done:
+                state = self.env.reset()
+            if T % self.args.replay_frequency == 0:
+                self.dqn.reset_noise()  # Draw a new set of noisy weights
+            action = self.dqn.act(state)  # Choose an action greedily (with noisy weights)
+            next_state, reward, done = self.env.step(action)  # Step
+            if self.args.reward_clip > 0:
+                reward = max(min(reward, self.args.reward_clip), -self.args.reward_clip)  # Clip rewards
+            self.mem.append(state, action, reward, done)  # Append transition to memory
+            if T >= self.args.learn_start:
+                self.mem.priority_weight = min(self.mem.priority_weight + priority_weight_increase, 1)  # Anneal importance sampling weight β to 1
+                if T % self.args.replay_frequency == 0:
+                    dqn.learn(self.mem)  # Train with n-step distributional double-Q learning
```

Comment:
same question for all the stuff that's conditional on T.  

I guess there are 2 ways I could think to approach this:

(1)- simple way, but a little less faithful to the original benchmark:
  just remove all the conditions (or force them to True), so each call to train() does all of the optional stuff.
   This is 'worse' since it distorts/magnifies the importance of the less frequent parts, leading to inaccurate benchmarking
  
(2)- more faithful way:
  increase niter to the point where you are sure you're covering all the branches of the conditionals.  This is 'better' since it would more heavily weight the computations that happen more frequently in real training,
  
(0) - current way:
  if niter == 1, then the current code would skip some steps, which is another form of distortion 
  
I guess my vote would be to try (2) and see how slow it is.  If it's too slow, maybe lean towards (1) or consider increasing frequencies a bit but not all the way to 'every time', so it can be a compromise